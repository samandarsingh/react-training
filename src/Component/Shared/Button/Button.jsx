import React from "react";
import PropTypes from "prop-types";
import styles from "./Button.module.scss";
/**
 * this is button field component using in shared
 * @param {string} buttonType 
 * @param {string} buttonName
 * @param {string} buttonClass 
 * @param {func} handleChange
 * @param {bool} index
 * @param {bool} status
 * @param {bool} disabled
 * @returns element
 */ 
const Button = props => {
    const {
        buttonType,
        buttonName,
        buttonClass,    
        handleOnClick,
        index,
        status,
        disabled        
    } = props;
    return (        
        <button
            type={buttonType}
            name={buttonName}            
            className={styles[buttonClass]}    
            onClick={e => handleOnClick(e,index,status)}  
            disabled={disabled}               
        >     
            {buttonName}     
        </button>        
    );
}
// set default props
Button.defaultProps = {
    buttonType: "",
    buttonName: "",
    buttonClass: "",    
    handleOnClick:  () => {},
    index: "",
    status: true,
    disabled: false
}
// proptypes validation
Button.propTypes = {
    buttonType: PropTypes.string.isRequired,
    buttonName: PropTypes.string,
    buttonClass: PropTypes.string,    
    handleOnClick: PropTypes.func,
    status: PropTypes.bool,
    disabled: PropTypes.bool.isRequired
  };
export default Button;