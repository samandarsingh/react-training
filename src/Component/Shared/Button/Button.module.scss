.buttonClass {
    padding: 10px 20px;
    background-color: dodgerblue;
    border: 1px solid #ddd;
    color: white;
    cursor: pointer;
}
.success {
    background: rgb(28, 184, 65); /* this is a green */
}
.error {
    background: rgb(202, 60, 60); /* this is a maroon */
}
.warning {
    background: rgb(223, 117, 20); /* this is an orange */
}
.secondary {
    background: rgb(66, 184, 221); /* this is a light blue */
}
.statusButton {
    background-color: white; 
    color: black; 
    border: 2px solid #008CBA;
}
.declineButton {
    background-color: white; 
    color: black; 
    border: 2px solid #f44336;
}
.isCompletedButton {
    background-color: white; 
    color: black; 
    border: 2px solid #4CAF50;
}