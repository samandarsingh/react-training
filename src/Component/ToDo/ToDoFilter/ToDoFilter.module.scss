.todoFilterContainer {
    margin-top: 50px;
}