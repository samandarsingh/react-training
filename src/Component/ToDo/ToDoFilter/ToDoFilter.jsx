import React from "react";
import PropTypes from "prop-types";
import styles from "./ToDoFilter.module.scss";
import Button from "../../Shared/Button";
/**
 * 
 * @param {fun} handleOnClick
 * @param {boolean} activeFilter
 * @returns Element
 */
const ToDoFilter = props => {
    const {handleOnClick, activeFilter} =props;
    return (
        <div className={styles.todoFilterContainer}>
            <Button 
                buttonType="button"
                buttonName="All"
                buttonClass={ activeFilter===0 ? "isCompletedButton":"statusButton"}
                handleOnClick={handleOnClick}    
                index={0} 
                status={ activeFilter===0 ? true : false} 
                disabled={ activeFilter===0 ? true : false}                        
            />
            &nbsp;
            <Button 
                buttonType="button"
                buttonName="Active"
                buttonClass={ activeFilter===1 ? "isCompletedButton":"statusButton"}
                handleOnClick={handleOnClick}    
                index={1} 
                status={ activeFilter===1 ? true : false} 
                disabled={ activeFilter===1 ? true : false}                       
            />
            &nbsp;
            <Button 
                buttonType="button"
                buttonName="Completed"
                buttonClass={ activeFilter===2 ? "isCompletedButton":"statusButton"}
                handleOnClick={handleOnClick}    
                index={2} 
                status={ activeFilter===2 ? true : false} 
                disabled={ activeFilter===2 ? true : false}                       
            />
        </div>
    );
}
// default props value
ToDoFilter.defaultProps = {
    handleOnClick: () => {},
    activeFilter: 0
  };
ToDoFilter.propstypes = {
    handleOnClick: PropTypes.func.isRequired,
    activeFilter: PropTypes.number.isRequired
}
export default ToDoFilter;