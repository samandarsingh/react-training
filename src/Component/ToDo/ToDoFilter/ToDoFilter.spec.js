import React from "react";
import { mount } from "enzyme";
import ToDoFilter from "./ToDoFilter";
import Button from "../../Shared/Button";
describe(`The ToDoFilter component testing`, () => { 
    let wrapper = "";
    beforeAll(() => {
        const props = {
            handleOnClick: () => {},
            filterActive: 0
        };
        wrapper = mount(<ToDoFilter {...props} />); 
    });
    it(`ToDoFilter component should be render without error`, () => {
        expect(wrapper).toBeTruthy();        
    });
    it(`ToDoFilter length should be equal to 3`,() => {
        expect(wrapper.find(`button`)).toHaveLength(3);
    });
    it(`ToDoFilter component get all records filter button testing`,() => {
        wrapper
        .find(`button`)
        .at(0)
        .simulate(`click`);
        expect(
            wrapper
            .find(Button)
            .at(0)
            .props()
            .buttonName
        )
        .toEqual(`All`);                
    });
    it(`ToDoFilter component get active records filter button testing`, () => {
        wrapper
        .find(`button`)
        .at(1)
        .simulate(`click`);
        expect(
            wrapper
            .find(Button)
            .at(1)
            .props()
            .buttonName
        )
        .toEqual(`Active`);        
    });
    it(`ToDoFilter component get completed records filter button testing`, () => {
        wrapper
        .find(`button`)
        .at(2)
        .simulate(`click`);
        expect(
            wrapper
            .find(Button)
            .at(2)
            .props()
            .buttonName
        )
        .toEqual(`Completed`);        
    });
    it(`SnapShot Testing`, () => {
        expect(wrapper).toMatchSnapshot();
    });
});