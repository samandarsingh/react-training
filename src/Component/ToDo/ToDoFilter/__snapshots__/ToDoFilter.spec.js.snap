// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ToDoFilter component testing SnapShot Testing 1`] = `ReactWrapper {}`;
